tests/tracer/test_sampler.py

```diff
@@ -689,7 +712,9 @@ def test_datadog_sampler_sample_no_rules(mock_sample, dummy_tracer):
         pass
     spans = dummy_tracer.pop()
     assert len(spans) == 1, "Span should have been written"
-    assert spans[0].get_metric(SAMPLING_PRIORITY_KEY) is AUTO_KEEP
+    assert_sampling_decision_tags(
+        spans[0], agent=1.0, limit=None, rule=None, sampling_priority=AUTO_KEEP, trace_tag="-0"
```

Comment:
I think this is mentioned in a comment by Tahir, but what is the expected behavior for the first payload sent to the agent, we don't yet have a response from the agent for it to tell us how to sample do we? In that case should the value still be `agent=1.0` ?

Reply:
Good question. The check for "agent" here is on the `_dd.agent_psr` metric on the root span, which defaults here to 1.0 being no rules or limit has been set. The existing test changed here did not include an assert on the this metric but for completeness it has been added here.